### ISSUE TITLE
Test cases fails with message "input is invalid type"

### ISSUE THREAD

**Issue by wobo-sumitha** [AOHUA/redux-state-sync#121]
Test cases written in React testing library is failing with the following error
` at Keccak.Object.<anonymous>.Keccak.update (node_modules/js-sha3/src/sha3.js:204:15)
      at node_modules/js-sha3/src/sha3.js:63:46
      at getPaths (node_modules/broadcast-channel/src/methods/node.js:59:29)
      at Object.create (node_modules/broadcast-channel/src/methods/node.js:367:19)
      at _prepareChannel (node_modules/broadcast-channel/dist/lib/broadcast-channel.js:210:37)
      at new BroadcastChannel (node_modules/broadcast-channel/dist/lib/broadcast-channel.js:62:3)
      at createStateSyncMiddleware (node_modules/redux-state-sync/dist/syncState.js:131:19)
      at Object.<anonymous> (src/redux/store.ts:8:19)
      at Object.<anonymous> (src/__tests__/test-util.jsx:6:1)
      at Object.<anonymous> (src/components/resusablecomponents/popup/Popup.test.tsx:5:1)`
      
      
![image](https://user-images.githubusercontent.com/86005800/138448767-ae4e2d21-4833-4627-a699-4d63035b317b.png)

**Comment by Boonstra** [comment]
I ran into the same problem. It seems to be because createSyncMiddleware will run as usual on state update, which it apparently can't in a test. I've solved the problem by mocking the functions I needed from the redux-state-sync library in the file where I'm setting up my Redux mock store:

```
jest.mock('redux-state-sync', () => ({
  createStateSyncMiddleware:
    () =>
    () =>
    (next: (action: PayloadAction) => void) =>
    (action: PayloadAction) =>
      next(action),
  initMessageListener: () => jest.fn(),
}));
```

Code is in TypeScript.

**Comment by aslam-fw** [comment]
I mocked the createStateSyncMiddleware in my test case. Even then facing the same issue.
<img width="507" alt="image" src="https://github.com/aohua/redux-state-sync/assets/38006143/58108216-3984-49f4-9c7e-869a06be00bb">

**Comment by reShambles** [comment]
_tl:dr;_

Provide your custom config with a channel key:

From:

**`const config: any = {}`**

To:

**`config = {
    channel: "yourChannelName"
}`**

**const stateSync = createStateSyncMiddleware(config);**

-------------------------------------------------------------------------------
_Long version:_


I may be late to the party but I ran into the same, at least similar problem. 

Everything worked fine in development and production but every test case with redux failed. 


The problem is your custom config - or to be more precise: the way the function

`createSyncMiddleware()` 

handles the custom config. 


If you do not specify a custom config, the library uses a custom one. This gets replaced once you fill the parameter with your 

`const config: any = {}`

The problem now occurs because it NEEDS a channel name as key - value.

`config = {
    channel: "yourChannelName"
}`

The way the function is implemented is by replacing and not merging the config you provide. So you override the default channel name with literally `undefined`. 

The way the browser handles this issue is, that it automatically casts undefined -> "undefined". So your channel name is

`config = {
    channel: "undefined"
}`

=> everything works. 

For test cases it uses server-side code (node by default in my case). Node did NOT cast undefined to a string and therefore was literally undefined which led to the error. 

**Solution 1 (instant):** 

Give your custom config a channel key (like mentioned above). At the time of writing I was at version 3.1.4 and the custom config looked like this: (found in the node package - syncState.js - line 9)

`const defaultConfig = {
    channel: 'redux_state_sync',
    predicate: null,
    blacklist: [],
    whitelist: [],
    broadcastChannelOption: undefined,
    prepareState: state => state,
    receiveState: (prevState, nextState) => nextState,
};`

Pick what you need from it (but at least the channel) and then set this as parameter like you did before. 

**Solution 2 (needs a new version):** 

Talk to the developer to not replace the custom config provided by the user but merge it with the default config (for example with a spread operator). 


Hope this helps.
